staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go

```diff
@@ -124,3 +124,41 @@ func New(lockType string, ns string, name string, coreClient corev1.CoreV1Interf
 		return nil, fmt.Errorf("Invalid lock-type %s", lockType)
 	}
 }
+
+// Manufacture will create a lock of a given type according to the input parameters
+func NewWithLabels(lockType string, ns string, name string, labels map[string]string, coreClient corev1.CoreV1Interface, coordinationClient coordinationv1.CoordinationV1Interface, rlc ResourceLockConfig) (Interface, error) {
```

Comment:
I found out this is no package calling this function at all, so what is the reason for adding and exporting this function? If it is necessary to make it public, we need more unit tests, but it does not seem necessary to me for now.

Reply:
Hi, I tried to add some unit test for this, but I found there is no test for `New` neither. 
Adding labels to the lock resources can be useful in other controller progress when the controller need to watch the specific lock resources, and reconcile to generate some other resources.
Considering such case, if we have a component with the ability of horizontal autoscaling, it means we have multiple instance for one component, then each instance will acquire one different lock resource,(each lock resource would usually named with the same name prefix). Currently, we can only filter the lock resource my `namespaces/name`,  no way to select by name prefix, so I consider to support adding lables to each lock resources.

Reply:
SGTM, but `New` function could call `NewWithLabels` with blank labels map to reduce the duplicate logic, and I do believe it would be great to have more unit tests.